tuned/plugins/plugin_scheduler.py

```diff
@@ -1065,3 +1075,99 @@ def _isolated_cores(self, enabling, value, verify, ignore_missing):
 			# Restoring processes' affinity is done in
 			# _instance_unapply_static()
 			self._restore_all_irq_affinity()
+
+	def _get_sched_knob_path(self, namespace, knob):
+		path = "/sys/kernel/debug/%s/%s" % (namespace, knob)
+		if not os.path.exists(path):
+			path = "/proc/sys/kernel/%s_%s" % (namespace, knob)
+		return path
+
+	def _get_sched_knob(self, namespace, knob):
+		return self._cmd.read_file(self._get_sched_knob_path(namespace, knob), err_ret = None)
+
+	def _set_sched_knob(self, namespace, knob, value, sim):
+		if value is None:
+			return None
+		if not sim:
+			self._cmd.write_to_file(self._get_sched_knob_path(namespace, knob), value)
+		return value
+
+	@command_get("sched_min_granularity_ns")
+	def _get_sched_min_granularity_ns(self):
+		return self._get_sched_knob("sched", "min_granularity_ns")
+
+	@command_set("sched_min_granularity_ns")
+	def _set_sched_min_granularity_ns(self, value, sim):
+		return self._set_sched_knob("sched", "min_granularity_ns", value, sim)
+
+	@command_get("sched_latency_ns")
+	def _get_sched_latency_ns(self):
+		return self._get_sched_knob("sched", "latency_ns")
+
+	@command_set("sched_latency_ns")
+	def _set_sched_latency_ns(self, value, sim):
+		return self._set_sched_knob("sched", "latency_ns", value, sim)
+
+	@command_get("sched_wakeup_granularity_ns")
+	def _get_sched_wakeup_granularity_ns(self):
+		return self._get_sched_knob("sched", "wakeup_granularity_ns")
+
+	@command_set("sched_wakeup_granularity_ns")
+	def _set_sched_wakeup_granularity_ns(self, value, sim):
+		return self._set_sched_knob("sched", "wakeup_granularity_ns", value, sim)
+
+	@command_get("sched_tunable_scaling")
+	def _get_sched_tunable_scaling(self):
+		return self._get_sched_knob("sched", "tunable_scaling")
+
+	@command_set("sched_tunable_scaling")
+	def _set_sched_tunable_scaling(self, value, sim):
+		return self._set_sched_knob("sched", "tunable_scaling", value, sim)
+
+	@command_get("sched_migration_cost_ns")
+	def _get_sched_migration_cost_ns(self):
+		return self._get_sched_knob("sched", "migration_cost_ns")
+
+	@command_set("sched_migration_cost_ns")
+	def _set_sched_migration_cost_ns(self, value, sim):
+		return self._set_sched_knob("sched", "migration_cost_ns", value, sim)
+
+	@command_get("sched_nr_migrate")
+	def _get_sched_nr_migrate(self):
+		return self._get_sched_knob("sched", "nr_migrate")
+
+	@command_set("sched_nr_migrate")
+	def _set_sched_nr_migrate(self, value, sim):
+		return self._set_sched_knob("sched", "nr_migrate", value, sim)
+
+	@command_get("numa_balancing_scan_delay_ms")
+	def _get_numa_balancing_scan_delay_ms(self):
+		return self._get_sched_knob("numa", "balancing_scan_delay_ms")
```

Comment:
This will create path "/sys/kernel/debug/numa/balancing_scan_delay_ms", but path in BZ (and also on system I tried) is "/sys/kernel/debug/sched/numa_balancing/scan_delay_ms". Same for other numa*.

Reply:
Good catch.